lib/Injector.php

```diff
@@ -369,6 +404,12 @@ public function make($name, array $args = array())
                 $reflectionFunction = $executable->getCallableReflection();
                 $args = $this->provisionFuncArgs($reflectionFunction, $args, null, $className);
                 $obj = call_user_func_array(array($executable, '__invoke'), $args);
+            } elseif (isset($this->proxies[$normalizedClass])) {
+                if (isset($this->prepares[$normalizedClass])) {
+                    $this->prepares_proxy[$normalizedClass] = $this->prepares[$normalizedClass];
+                }
+                $obj = $this->resolveProxy($className, $normalizedClass, $args);
+                unset($this->prepares[$normalizedClass]);
```

Comment:
Why are those unset?

Reply:
Because preparation is made inside the `resolveProxy()' with 'prepareInstance()' when the instantiation of the real object is called at runtime, this happens after the 'make()' method is called, those unset make sure that the 'prepareInstance()' called on the line 417 will be bypassed without any error.